src/main/java/com/hubspot/jinjava/tree/parse/TokenScannerSymbols.java

```diff
@@ -15,24 +15,86 @@
  **********************************************************************/
 package com.hubspot.jinjava.tree.parse;
 
-public interface TokenScannerSymbols {
-  char TOKEN_PREFIX_CHAR = '{';
-  char TOKEN_POSTFIX_CHAR = '}';
-  char TOKEN_FIXED_CHAR = 0;
-  char TOKEN_NOTE_CHAR = '#';
-  char TOKEN_TAG_CHAR = '%';
-  char TOKEN_EXPR_START_CHAR = '{';
-  char TOKEN_EXPR_END_CHAR = '}';
-  char TOKEN_NEWLINE_CHAR = '\n';
-  char TOKEN_TRIM_CHAR = '-';
-
-  int TOKEN_PREFIX = TOKEN_PREFIX_CHAR;
-  int TOKEN_POSTFIX = TOKEN_POSTFIX_CHAR;
-  int TOKEN_FIXED = TOKEN_FIXED_CHAR;
-  int TOKEN_NOTE = TOKEN_NOTE_CHAR;
-  int TOKEN_TAG = TOKEN_TAG_CHAR;
-  int TOKEN_EXPR_START = TOKEN_EXPR_START_CHAR;
-  int TOKEN_EXPR_END = TOKEN_EXPR_END_CHAR;
-  int TOKEN_NEWLINE = TOKEN_NEWLINE_CHAR;
-  int TOKEN_TRIM = TOKEN_TRIM_CHAR;
+public abstract class TokenScannerSymbols {
+  private String expressionStart = null;
+  private String expressionStartWithTag = null;
+  private String closingComment = null;
+
+  public abstract char getPrefixChar();
+
+  public abstract char getPostfixChar();
+
+  public abstract char getFixedChar();
+
+  public abstract char getNoteChar();
+
+  public abstract char getTagChar();
+
+  public abstract char getExprStartChar();
+
+  public abstract char getExprEndChar();
+
+  public abstract char getNewlineChar();
+
+  public abstract char getTrimChar();
+
+  public int getPrefix() {
```

Comment:
why are all these methods that just call abstract methods needed?

Reply:
there are places when an integer is being required and there are separate places where a character is required during parsing the template. Also, it will be better if the user is allowed to fill in the functions which return `char` and not the ASCII value instead. Somebody who wants custom tokens will only have to fill in the tokens in form of characters and do not need to write down their ASCII values.
Also, I could've used those abstract methods instead since they can be cast to integers as well but the problem lies with the fact that when I was writing the `if..else` conditions then it was becoming a lot unclear since the casting had to be done during the comparison. Therefore I thought It'd be better to not mess up the code readability a lot and simply have users fill out the functions which return `char` and then return their `int` value using the `getPrefix` functions instead. I believe it was a better way to make changes.
I hope my explanation is fine. If you feel that it is not okay then I'd try and make the changes.